### ISSUE TITLE
Body can be null

### ISSUE THREAD

**Issue by koenhoeijmakers** [npm/npm-registry-fetch#6]
We're getting an error in our pipeline that references to the following:
```
npm timing stage:rollbackFailedOptional Completed in 1ms
npm timing stage:runTopLevelLifecycles Completed in 4666ms
npm verb stack TypeError: Cannot read property 'toString' of null
npm verb stack     at /usr/local/lib/node_modules/npm/node_modules/npm-registry-fetch/check-response.js:98:66
```

Which is `body.toString('utf-8')`, i see that there's another `body.toString('utf-8')` higher up that is being try catched, shouldn't that also happen here?

**Comment by blackstar257** [comment]
Just ran into this as well.

**Comment by koenhoeijmakers** [comment]
I did find out why it was throwing this error for me. 

It was due to NPM trying to load a package that was in a private repo (fontawesome pro) and because i forgot to set the token in my config it resulted in the body being null (and thus throwing this error).

But i'm pretty certain that it should not throw this TypeError.

**Comment by blackstar257** [comment]
Same here it ended up being my token for fontawesome got regenerated causing it to 401. i was able to view the error verbosely by running `ncu`

**Comment by blackstar257** [comment]
The error message though wasn't helpful though so i think this issue is still valid but maybe renamed to something like update error handling message when request isn't what is expected

**Comment by isaacs** [comment]
Closed by #9

**Comment by yaronlevi** [comment]
This thread got me out of dead-end trying to figure out why a Bitrise build is failing.
It was a private NPM repo that was not accessible due to the expiry of an auth token.
Saved me many days 🙏🏼
